### PR TITLE
sink(ticdc): add flush worker to flush row changed events

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/4423

### What is changed and how it works?

add flush worker to flush row changed events.

Main processes:
1. Gather messages (multiple partition messages are collected)
2. Classified by parition
3. The events are encoded and sent to the producer by partition.

### Check List 

Tests 

 - [x] Unit test
 - [x] benchmark

Code changes

None

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/hold 

base on https://github.com/pingcap/tiflow/pull/4630

**Comment by hi-rustin** [comment]
/uncc asddongmen 
/cc @overvenus 

Because asddongmen on leave, so please help to see.

**Comment by hi-rustin** [comment]
> I'm sorry I tested with my own computer, but currently the runc runtime on tispace is not working properly due to some problems. idc's machine is heavily oversold. So maybe my local machine is an acceptable option.

Let's do a more realistic performance test before GA, but this current result should also be informative.

File: cdc/sink/mq_flush_worker.go
Comment (overvenus) [inline cdc/sink/mq_flush_worker.go:84]: Could you reuse the tick? `NewTicker` spawns a goroutine every time.

File: cdc/sink/mq_flush_worker.go
Comment (overvenus) [inline cdc/sink/mq_flush_worker.go:102]: Not sure if it helps performance or lantency, could you test high throughput  and low throughput scenario later?

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: cdc/sink/mq_flush_worker.go
Comment (hi-rustin) [reply cdc/sink/mq_flush_worker.go]: Done.

File: cdc/sink/mq_flush_worker.go
Comment (hi-rustin) [reply cdc/sink/mq_flush_worker.go]: I think it might help with latency at low loads, and I'll follow up by looking at it when I do real-world performance testing.

File: cdc/sink/mq_flush_worker.go
Comment (asddongmen) [inline cdc/sink/mq_flush_worker.go:83]: Would it be more appropriate to use timer here? 
ref: https://gobyexample.com/tickers#:~:text=Timers%20are%20for%20when%20you,periodically%20until%20we%20stop%20it.&text=Tickers%20can%20be%20stopped%20like%20timers.

**Comment by hi-rustin** [comment]
> /LGTM

@zhaoxinyu  You need to use GitHub approve feature :D

File: cdc/sink/mq_flush_worker.go
Comment (asddongmen) [reply cdc/sink/mq_flush_worker.go:83]: LGTM after reuse this ticker.

**Review by asddongmen — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**
